src/main/webapp/js/chart-loader.js

```diff
@@ -79,18 +112,17 @@
     chart.draw(diffFavoriteTea, pieOptions);
   };
 
-
   var init = function() {
     google.charts.load("current", {
       packages: ["corechart"],
     });
     google.charts.setOnLoadCallback(
       function() {
+        fetchMessageData();
```

Comment:
Retrieving the message can happen sooner. You only need to wait for the Google Chart's onload event for using the types defined by that API. You should be able to initiate the backend `fetch` directly within `init`.

Reply:
Okay, I just want to see if I am understanding this right now after reading up on everything again and combing through the past comments as well.

So, when chart-loader.js is running, it’s going to first run everything that is in ```init()``` first and this is where want to call ```fetchMessageData()``` so that we fetch our data first. Then, with the help of ```.setOnLoadCallback()``` , it will call ```drawChart()``` after we finish calling ```init()```. This will prevent the potential run condition that we are going to have because we are getting the chart data before drawing the chart. Not like before, where we are drawing the chart and fetching the message data at the same time because the code we had before was asynchronous.  

If this is right, then all of this makes more sense now.